addons/account/models/account_move.py

```diff
@@ -3786,8 +3786,9 @@ def _reverse_moves(self, default_values_list=None, cancel=False):
 
     def _can_be_unlinked(self):
         self.ensure_one()
-        lock_date = self.company_id._get_user_fiscal_lock_date()
-        return not self.inalterable_hash and self.date > lock_date
+        lock_date = self.company_id._get_user_fiscal_lock_date(self.journal_id)
```

Comment:
:copyright: :spaghetti: 

```suggestion
        lock_date = self.company_id._get_user_fiscal_lock_date()
```

Reply:
Indeed :man_facepalming: :sweat_smile: 
And there is another issue that I'm trying to fix, a `UserError` that says: "The remaining value on the last depreciation line must be 0"

Reply:
That might be because `asset_id` is copied when reversing, so it is part of the depreciation schedule; but we don't want that in this case?

I might be confused between different versions